src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -37,12 +38,12 @@ internal class MgmtOutputLibrary : OutputLibrary
         /// <summary>
         /// This is a map from raw request path to the corresponding <see cref="Resource"/>
         /// </summary>
-        private Dictionary<string, Resource>? _rawRequestPathToArmResource;
+        private Dictionary<string, Dictionary<ResourceType, Resource>>? _rawRequestPathToArmResource;
```

Comment:
Lets discuss this, its not intuitive why all of these types need a dictionary now and I wonder if there is a more clear way to represent this